app/pulp/app/models/progress.py

```diff
@@ -18,6 +20,16 @@ class ProgressReport(Model):
 
     All progress reports have a message, state, and are related to a Task.
 
+    Attributes:
+
+        batch_increments (bool): If this Flag is set save checks to make sure it only gets ran once
```

Comment:
I see now that this is being used to ensure it's only run during the context manager. That is good (and required), so converting it to a private variable and calling it `using_context_manager` is what I recommend.